### DISCUSSION TITLE
Common config to share across multiple micro services

### DISCUSSION THREAD

**Discussion by tommy38hk** [hyness/spring-cloud-config-server#84] [Q&A]
Hi,
I would like to create a common config to share between multiple micro services

I have the following define in my micro service application.yml

spring:
  cloud:
    config:
      profile: auth0
      enabled: true
      name: common-settings, ${spring.application.name}
  config:
    import: configserver:http://localhost:8881/

However, it does not seem to be able to read the config from the common-settings.yml file in the config server

My docker-compose is as follows

services:
  config:
    container_name: ${USERNAME}-config
    hostname: ${USERNAME}-config
    image: "hyness/spring-cloud-config-server:jdk8"
    privileged: true
    shm_size: 2g
    tty: true
    ports:
      - "8881:8888"
    environment:
      SPRING_PROFILES_ACTIVE: native
    volumes:
      - ./config:/config

Is there a search-path config that I need to define?

**Comment by hyness** [comment]
Hello @tommy38hk,

It's hard to help based on the information supplied.  The way you have configured cloud config in your `application.yml`, the app shouldn't be able to start if the config server is inaccessible.  You didn't mention the app not starting, so I'll assume it did.  Typically this happens when the microservice doesn't include the cloud config client jar, which can be found in [maven central](https://search.maven.org/artifact/org.springframework.cloud/spring-cloud-config-client/3.1.1/jar).  You need that so your microservice can talk to cloud config.  If you have that jar, I'd check to make sure the server is accessible by curling the repo.

```
curl -v http://localhost:8881/common-settings/default
```

Hard to tell based on formatting, but be sure to set the following as your application name
```
spring.application.name=common-settings
```

Be sure to read the section on [Spring Cloud Config Client](https://cloud.spring.io/spring-cloud-config/reference/html/#_spring_cloud_config_client) in the Spring documentation

Hope this helps!

**Comment by tommy38hk** [comment]
The application can be started just that I am unable to get the config from the config server.  curl -v http://localhost:8881/common-settings/default did give a result.
I have two config files, one is common-settings.yml and the other one is xxx which is my ${spring.application.name}
I am trying to share commons-settings between the micro services hence I have put two names in the name config

name: common-settings, ${spring.application.name}

If I merge the common-settings into the xxx.yml that I can see the values from my application

From the Cloud config server console, I could see both files were loaded

**Reply by hyness** [reply]
Instead of using `common-settings.yml`, try renaming it to `application.yml`, which tells Spring Cloud Config Server to add the properties to every project.  AFAIK, you can't have multiple application names.  You can have multiple profiles under the application name

**Reply by tommy38hk** [reply]
I tried putting the application.yml to the config folder but still did not work

In the following discussion, it looks like I need override the application properties in the config server to add the search path?

https://stackoverflow.com/questions/55232105/spring-cloud-config-server-how-to-override-shared-properties-in-app-specific-fi
